python/ray/data/read_api.py

```diff
@@ -1470,6 +1471,113 @@ def read_text(
     )
 
 
+@PublicAPI
+def read_avro(
+    paths: Union[str, List[str]],
+    *,
+    encoding: str = "utf-8",
+    drop_empty_lines: bool = True,
+    filesystem: Optional["pyarrow.fs.FileSystem"] = None,
+    parallelism: int = -1,
+    ray_remote_args: Optional[Dict[str, Any]] = None,
+    arrow_open_stream_args: Optional[Dict[str, Any]] = None,
+    meta_provider: Optional[BaseFileMetadataProvider] = None,
+    partition_filter: Optional[PathPartitionFilter] = None,
+    partitioning: Partitioning = None,
+    include_paths: bool = False,
+    ignore_missing_paths: bool = False,
+    shuffle: Union[Literal["files"], None] = None,
+    file_extensions: Optional[List[str]] = None,
+    concurrency: Optional[int] = None,
+    override_num_blocks: Optional[int] = None,
+) -> Dataset:
+    """Create a :class:`~ray.data.Dataset` from records stored in avro files.
+
+    Examples:
+        Read a avro file in remote storage or local storage.
+
+        >>> import ray
+        >>> ds = ray.data.read_avro("s3://anonymous@ray-example-data/mnist.avro")
```

Comment:
This file won't exist, but I can add it

Reply:
That would be great, please go ahead if it's separate from the PR. I also wonder how to add it? Do I have access?

Reply:
OK I figured it out, it's actually under the the path: `doc/source/data/examples`
I just updated it and let's see if the test will pass.